eventlet/support/greendns.py

```diff
@@ -120,6 +120,15 @@ def is_ip_addr(host):
     return is_ipv4_addr(host) or is_ipv6_addr(host)
 
 
+def compute_expiration(query, timeout):
+    # NOTE(ralonsoh): in dnspython v2.0.0, "_compute_expiration" was replaced
+    # by "_compute_times".
+    if hasattr(query, '_compute_expiration'):
```

Comment:
Think you can do this attribute check at module level to avoid runtime cost?

Reply:
Updated too :)